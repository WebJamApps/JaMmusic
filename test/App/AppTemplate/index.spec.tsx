import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import {
  AppTemplate, handleEscapePress, handleKeyMenu, makeDrawerClass,
} from 'src/App/AppTemplate';

describe('AppTemplate', () => {
  it('renders the component', () => {
    const props = {
      dispatch: jest.fn(),
      auth: {
        isAuthenticated: false, token: '', error: '', user: { userType: '', email: '' },
      },
      userCount: 0,
      heartBeat: 'white',
      history: {} as any,
      location: { pathname: '/' } as any,
      match: {} as any,
    };
    const at = renderer.create(
      <BrowserRouter><AppTemplate {...props}><div id="test-page" /></AppTemplate></BrowserRouter>,
    ).toJSON();
    expect(at).toMatchSnapshot();
  });
  it('makeDrawerClass', () => {
    expect(makeDrawerClass(true)).toBe('home-sidebar open drawer-container');
  });
  it('handleEscapePress', () => {
    const setMenuOpen = jest.fn();
    handleEscapePress({ key: 'Escape' }, setMenuOpen);
    expect(setMenuOpen).toHaveBeenCalledWith(false);
  });
  it('handleKeyMenu', () => {
    const setMenuOpen = jest.fn();
    handleKeyMenu({ key: 'Enter' }, false, setMenuOpen);
    expect(setMenuOpen).toHaveBeenCalledWith(true);
  });
});
