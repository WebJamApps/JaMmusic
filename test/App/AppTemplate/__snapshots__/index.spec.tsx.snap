// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppTemplate renders the component 1`] = `
<div
  className="page-host"
>
  <div
    className="home-sidebar close drawer-container"
    id="sidebar"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="drawer"
      style={
        {
          "backgroundColor": "#c0c0c0",
          "position": "relative",
          "zIndex": -1,
        }
      }
    >
      <div
        className="navImage"
      >
        <img
          alt="wjsidelogo"
          id="webjamwidelogo"
          src="../static/imgs/webjamlogo1.png"
          style={
            {
              "marginLeft": 0,
              "marginRight": 0,
              "width": "182px",
            }
          }
        />
      </div>
      <div
        className="nav-list"
        style={
          {
            "width": "180px",
          }
        }
      >
        <div
          id="musTT"
          style={
            {
              "backgroundColor": "white",
              "display": "none",
              "padding": "3px",
              "position": "absolute",
              "right": "68px",
              "top": "305px",
            }
          }
        >
          Music
        </div>
        <div
          className="menu-item"
        >
          <a
            className="nav-link"
            href="/music"
            onClick={[Function]}
          >
            <div
              style={
                {
                  "display": "inline",
                }
              }
            >
              <i
                className="fas fa-music"
                style={
                  {
                    "marginRight": "8px",
                  }
                }
              />
              <span
                className="nav-item"
              >
                Music
              </span>
            </div>
          </a>
        </div>
        <div
          className="menu-item"
        >
          <a
            className="nav-link"
            href="/"
            onClick={[Function]}
          >
            <div
              style={
                {
                  "display": "inline",
                }
              }
            >
              <i
                className="fas fa-home"
                style={
                  {
                    "marginRight": "8px",
                  }
                }
              />
              <span
                className="nav-item"
              >
                Web Jam LLC
              </span>
            </div>
          </a>
        </div>
        <div
          className="menu-item googleLogin"
        >
          <button
            className="loginButton"
            onClick={[Function]}
            size="small"
            variant="contained"
          >
            Login
          </button>
        </div>
        <p
          style={
            {
              "fontSize": "6pt",
              "margin": 0,
              "padding": 0,
            }
          }
        >
          Â 
        </p>
        <div
          className="active-users"
        >
          <div />
        </div>
      </div>
    </div>
  </div>
  <div
    className="main-panel"
  >
    <span
      id="mobilemenutoggle"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <i
        className="fas fa-bars"
      />
    </span>
    <div
      className="mainPanel"
    >
      <div
        className="swipe-area"
      />
      <div
        className="material-header home-header"
        id="header"
      >
        <div
          className="headercontent"
          id="ohaflogo"
        >
          <img
            alt="ohaflogo"
            className="home-header-image"
            src="../static/imgs/webjamicon7.png"
          />
        </div>
        <div
          className="headercontent header-text-card"
        >
          <h3
            className="header-text"
            style={
              {
                "marginTop": 0,
              }
            }
          >
            Web Jam LLC
          </h3>
        </div>
      </div>
      <div
        className="content-block"
        id="contentBlock"
        style={
          {
            "width": "auto",
          }
        }
      >
        <div
          id="test-page"
        />
        <div
          className="footer"
          id="wjfooter"
          style={
            {
              "backgroundColor": "#565656",
              "bottom": "0",
              "paddingBottom": "20px",
              "paddingTop": "20px",
            }
          }
        >
          <div
            style={
              {
                "padding": "6px",
                "textAlign": "center",
              }
            }
          >
            <a
              href="https://github.com/WebJamApps"
              rel="noopener noreferrer"
              style={
                {
                  "color": "#c09580",
                  "paddingRight": "5px",
                }
              }
              target="_blank"
            >
              <span>
                <i
                  className="fab fa-github"
                />
              </span>
            </a>
            <a
              href="https://www.linkedin.com/company/webjam/"
              rel="noopener noreferrer"
              style={
                {
                  "color": "#c09580",
                  "paddingRight": "5px",
                }
              }
              target="_blank"
            >
              <span>
                <i
                  className="fab fa-linkedin"
                />
              </span>
            </a>
            <a
              href="https://www.instagram.com/joshua.v.sherman/"
              rel="noopener noreferrer"
              style={
                {
                  "color": "#c09580",
                  "paddingRight": "5px",
                }
              }
              target="_blank"
            >
              <span>
                <i
                  className="fab fa-instagram"
                />
              </span>
            </a>
            <a
              href="https://twitter.com/JoshuaVSherman"
              rel="noopener noreferrer"
              style={
                {
                  "color": "#c09580",
                  "paddingRight": "5px",
                }
              }
              target="_blank"
            >
              <span>
                <i
                  className="fab fa-twitter"
                />
              </span>
            </a>
            <a
              href="https://www.facebook.com/WebJamLLC/"
              rel="noopener noreferrer"
              style={
                {
                  "color": "#c09580",
                  "paddingRight": "5px",
                }
              }
              target="_blank"
            >
              <span>
                <i
                  className="fab fa-facebook"
                />
              </span>
            </a>
            <a
              href="https://joshuavsherman.tumblr.com/"
              rel="noopener noreferrer"
              style={
                {
                  "color": "#c09580",
                  "paddingRight": "5px",
                }
              }
              target="_blank"
            >
              <span>
                <i
                  className="fab fa-tumblr"
                />
              </span>
            </a>
            <p
              style={
                {
                  "color": "white",
                  "fontSize": "9pt",
                  "marginBottom": 0,
                }
              }
            >
              Powered by
               
              <a
                className="wjllc"
                href="https://www.web-jam.com"
                rel="noopener noreferrer"
                target="_blank"
              >
                Web Jam LLC
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
